custom_components/e3dc_rscp/sensor.py

```diff
@@ -372,6 +372,92 @@
         device_class=SensorDeviceClass.POWER_FACTOR,
         state_class=SensorStateClass.MEASUREMENT,
     ),
+    SensorEntityDescription(
+        key="wallbox-app-software",
+        translation_key="wallbox-app-software",
+        icon="mdi:information-outline",
+        device_class=None,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="wallbox-consumption-net",
+        translation_key="wallbox-consumption-net",
+        icon="mdi:transmission-tower-import",
+        native_unit_of_measurement=UnitOfPower.WATT,
+        suggested_unit_of_measurement=UnitOfPower.KILO_WATT,
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key="wallbox-consumption-sun",
+        translation_key="wallbox-consumption-sun",
+        icon="mdi:solar-power",
+        native_unit_of_measurement=UnitOfPower.WATT,
+        suggested_unit_of_measurement=UnitOfPower.KILO_WATT,
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key="wallbox-energy-all",
+        translation_key="wallbox-energy-all",
+        icon="mdi:counter",
+        native_unit_of_measurement=UnitOfEnergy.WATT_HOUR,
+        suggested_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        suggested_display_precision=2,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    SensorEntityDescription(
+        key="wallbox-energy-net",
+        translation_key="wallbox-energy-net",
+        icon="mdi:counter",
+        native_unit_of_measurement=UnitOfEnergy.WATT_HOUR,
+        suggested_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        suggested_display_precision=2,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    SensorEntityDescription(
+        key="wallbox-energy-sun",
+        translation_key="wallbox-energy-sun",
+        icon="mdi:counter",
+        native_unit_of_measurement=UnitOfEnergy.WATT_HOUR,
+        suggested_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        suggested_display_precision=2,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    SensorEntityDescription(
```

Comment:
Would this be also a candidate for a diagnostics entity?

Reply:
Yes, will add this.

Reply:
yes, changed that